ObjectTK.Tools/Cameras/Camera.cs

```diff
@@ -7,116 +7,70 @@
 // of the MIT license. See the LICENSE file for details.
 //
 
+using ObjectTK.Tools.Mathematics;
+using OpenTK.Mathematics;
 using System;
-using OpenTK;
-using OpenTK.Input;
 
-namespace ObjectTK.Tools.Cameras
-{
-    public class Camera
-    {
-        public CameraState State;
-        public CameraState DefaultState;
-        protected CameraBehavior Behavior;
+namespace ObjectTK.Tools.Cameras {
+	public class Camera {
+		public Vector3 Position { get; set; } = Vector3.Zero;
+		public Quaternion Rotation { get; set; } = Quaternion.FromEulerAngles(new Vector3(0, (float)Math.PI, 0));
+		public Vector3 Forward => Vector3.Transform(Vector3.UnitZ, Rotation).Normalized();
+		public Vector3 Up => Vector3.UnitY;
+		public Vector3 Right => Vector3.Cross(Forward, Up);
+		public float FieldOfView { get; set; } = MathHelper.PiOver2;
+		public float AspectRatio => Viewport.Size.X / (float)Viewport.Size.Y;
+		public Matrix4 ViewMatrix => Matrix4.LookAt(Position, Position + Forward, Up);
+		public Matrix4 ProjectionMatrix => Matrix4.CreatePerspectiveFieldOfView(FieldOfView, AspectRatio, NearClippingPlaneDistance, FarClippingPlaneDistance);
+		public Matrix4 ViewProjectionMatrix => ViewMatrix * ProjectionMatrix;
+		public Box2i Viewport { get; set; }
 
-        public float MouseMoveSpeed = 0.005f;
-        public float MouseWheelSpeed = 0.1f;
-        public float MoveSpeed = 60;
+		public float NearClippingPlaneDistance => 0.1f;
+		public float FarClippingPlaneDistance => 1000.0f;
 
-        public Camera()
-        {
-            State = new CameraState();
-            DefaultState = new CameraState();
-        }
+		public Matrix4 GetCameraTransform() {
```

Comment:
Handle pitch clamping here with less of a hack than was previously implemented

Reply:
This function has no references anymore since I added the `ViewProjectionMatrix` property, just removing it. Also removing the `Rotate` and `LookAt` methods to accommodate different camera types people may want, didn't make sense to force `Up = Vector3.UnitY` 

At least in game engines I've seen the Camera has a Transform and camera state is managed some other way like holding 2 floats for pitch and yaw and constructing the rotation based on those instead of rotating directly, so I don't think it would be appropriate to store those in the camera class